plugins/java/connectors/imap-connector/src/main/java/com/lucidworks/fusion/connector/plugin/ImapFetcher.java

```diff
@@ -0,0 +1,76 @@
+package com.lucidworks.fusion.connector.plugin;
+
+import com.lucidworks.fusion.connector.plugin.api.fetcher.Fetcher;
+import com.lucidworks.fusion.connector.plugin.api.fetcher.context.FetchContext;
+import com.lucidworks.fusion.connector.plugin.api.fetcher.context.StopContext;
+import com.lucidworks.fusion.connector.plugin.client.Email;
+import com.lucidworks.fusion.connector.plugin.client.ImapClient;
+import org.apache.logging.log4j.Logger;
+
+import javax.inject.Inject;
+import javax.mail.MessagingException;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class ImapFetcher implements Fetcher {
+  private final Logger logger;
+  private final ImapConfig config;
+
+  private ImapClient imapClient;
+
+  @Inject
+  public ImapFetcher(
+      Logger logger,
+      ImapConfig config,
+      ImapClient imapClient
+  ) {
+    this.logger = logger;
+    this.config = config;
+    this.imapClient = imapClient;
+  }
+
+  @Override
+  public FetchResult fetch(FetchContext fetchContext) {
+    try {
+      for (Email message : imapClient.getUnreadMessages(this.config.getProperties().getFolder())) {
+        Map<String, Object> data = new HashMap<>();
+
+
+        data.put("id", message.getId());
```

Comment:
Please use Constants for `id`, `from`, `reply_to`, `to`, etc. You might create a class named `ImapConstants` and put these within it.